modules/skybrowser/src/screenspaceskytarget.cpp

```diff
@@ -0,0 +1,430 @@
+                _animationStart,
+                _animationEnd,
+                deltaTime,
+                _animationSpeed
+            );
+
+            // Rotate target direction
+            glm::dvec3 equatorial = glm::dvec3(rotMat * glm::dvec4(_animationStart, 1.0));
+            glm::dvec3 localCamera = skybrowser::equatorialToLocalCamera(equatorial);
+            // Convert to screen space
+            if (_useRadiusAzimuthElevation) {
+                // Keep the radius
+                glm::vec3 position = _raePosition.value().x * glm::vec3(localCamera);
+                _raePosition = cartesianToRae(position);
+            }
+            else {
+                _cartesianPosition = skybrowser::localCameraToScreenSpace3d(localCamera);
+            }
+            
+            // Update position
+            glm::dvec3 cartesian = skybrowser::sphericalToCartesian(equatorialAim());
+            _animationStart = glm::normalize(cartesian);
+        }
+        else {
+            // Set the exact target position 
+            glm::dvec3 localCamera = skybrowser::equatorialToLocalCamera(_animationEnd);
+          
+            if (_useRadiusAzimuthElevation) {
+                glm::vec3 position = _raePosition.value().x * glm::vec3(localCamera);
+                _raePosition = cartesianToRae(position);
+            }
+            else {
+                _cartesianPosition = skybrowser::localCameraToScreenSpace3d(localCamera);
+            }
+            
+            _isAnimated = false;
+            // Lock target when it first arrives to the position
+            setLock(_shouldLockAfterAnimation);
+        }      
+    }
+
+    glm::dvec2 ScreenSpaceSkyTarget::equatorialAim() const {
+        glm::dvec3 cartesian;
+        // Get the local camera coordinates of the target
+        if (_useRadiusAzimuthElevation) {
+            cartesian = raeToCartesian(_raePosition.value());
+            glm::dvec3 equatorial = skybrowser::localCameraToEquatorial(cartesian);
+            return skybrowser::cartesianToSpherical(equatorial);
+            
+        }
+        else {
+            cartesian = skybrowser::localCameraToEquatorial(_cartesianPosition.value());
+            return skybrowser::cartesianToSpherical(cartesian);
+        }
+    }
+
+    void ScreenSpaceSkyTarget::highlight(glm::ivec3 addition) {
+        _borderColor += addition;
+    }
+
+    void ScreenSpaceSkyTarget::removeHighlight(glm::ivec3 removal) {
+        _borderColor -= removal;
+    }
+
+    float ScreenSpaceSkyTarget::opacity() const {
+        return _opacity;
+    }
+
+    glm::dvec2 ScreenSpaceSkyTarget::lockedCoordinates() const {
+        return skybrowser::cartesianToSpherical(_lockedCoordinates);
+    }
+
+    void ScreenSpaceSkyTarget::setOpacity(float opacity) {
+        _opacity = opacity;
+    }
+
+    void ScreenSpaceSkyTarget::setLock(bool isLocked) {
+        _isLocked = isLocked;
+        if (_isLocked) {
```

Comment:
seems like this part should be in the "onChange" method of the _isLocked property. Is that correct?

Reply:
Oh sorry, makes total sense!